### PR TITLE
Static Todo list demo

### PR DESCRIPTION
**Pull request by danybony**
Display a static TODO list filled using items defined in resources string-array.
_Note:_ this disables the Twitter module, as the todo list is displayed in the center of the screen instead of it. It will be reenabled in a future PR to alternate the two demo (todo and alarm) and the twitter feed.
todo

### CODE REVIEW

File: app/src/main/java/net/bonysoft/magicmirror/MainActivity.java [hunk 1]
```diff
@@ -43,6 +46,7 @@
     private TextView tweetLabel;
     private TextView weatherTemperatureLabel;
     private TextView todayForecastLabel;
+    private ListView todoList;
```
Comment (alexstyl) [inline app/src/main/java/net/bonysoft/magicmirror/MainActivity.java:49]: `ListView`? So retro

File: app/src/main/java/net/bonysoft/magicmirror/todo/TodoItemsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import android.widget.CheckBox;
+
+import java.util.Arrays;
+import java.util.List;
+
+import net.bonysoft.magicmirror.R;
+
+public class TodoItemsAdapter extends ArrayAdapter<CharSequence> {
+
+    private List<CharSequence> todoItems;
```
Comment (alexstyl) [inline app/src/main/java/net/bonysoft/magicmirror/todo/TodoItemsAdapter.java]: `final`

File: app/src/main/java/net/bonysoft/magicmirror/todo/TodoItemsAdapter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    public TodoItemsAdapter(Context context, int textViewResourceId, CharSequence[] todoItems) {
+        super(context, textViewResourceId, todoItems);
+        this.todoItems = Arrays.asList(todoItems);
+    }
+
+    private class ViewHolder {
```
Comment (alexstyl) [inline app/src/main/java/net/bonysoft/magicmirror/todo/TodoItemsAdapter.java]: This should be `static`, no?

File: app/src/main/java/net/bonysoft/magicmirror/MainActivity.java (hunk 1 above)
Comment (hhaouat) [inline app/src/main/java/net/bonysoft/magicmirror/MainActivity.java:49]: Why not a recycler view? for example

Comment (danybony) [inline app/src/main/java/net/bonysoft/magicmirror/MainActivity.java:49]: Honestly I was tempted to not even use a `ListView` but just copy the items layout into a dedicated layout. This at least gives us a bit of flexibility, with just the need for a change in the resources file.

Regarding why not a `RecyclerView` the reason is that there is nothing to be recycled here, the number of items would be always contained and using a  `RecyclerView` would definitely be an overkill

**Comment by alexstyl** [comment]
👍 LGTM

File: app/src/main/java/net/bonysoft/magicmirror/MainActivity.java (hunk 1 above)
Comment (hhaouat) [inline app/src/main/java/net/bonysoft/magicmirror/MainActivity.java:49]: 👍

**Pull request merged**